core/device_impl.cpp

```diff
@@ -155,6 +159,50 @@ void DeviceImpl::process_autopilot_version(const mavlink_message_t &message)
     unregister_timeout_handler(_autopilot_version_timed_out_cookie);
 }
 
+void DeviceImpl::process_statustext(const mavlink_message_t &message)
+{
+    mavlink_statustext_t statustext;
+    mavlink_msg_statustext_decode(&message, &statustext);
+
+    std::string debug_str = "mavlink ";
```

Comment:
Is this enough to differentiate device as opposed to DroneCore originated warnings? Maybe text:
`MAVLink: emergency` or `Device: emergency`. 
Might even be worth having a special class of debug for these. Just a thought.

Reply:
Agreed @hamishwillee, it would be good to understand when are these called and if we should be raising exceptions or alerting the users rather than just using them for debugging.

That said this PR is to have more debugging data, so we should merge and get someone to research onto what the messages mean, I want to know if we should be taking more critical actions

Reply:
@hamishwillee I can change it to `MAVLink: emergency`.

@mrpollo to catch errors, it might make sense that the plugins check for errors and inform the API user if it is relevant.

Reply:
> @mrpollo That said this PR is to have more debugging data, so we should merge and get someone to research onto what the messages mean, I want to know if we should be taking more critical actions

> @mrpollo to catch errors, it might make sense that the plugins check for errors and inform the API user if it is relevant.

> @hamishwillee I can change it to MAVLink: emergency.

Agree with all above - yes, ideally plugins should check for errors and work out what is relevant but we need to do more research on this. IMO we should do change suggested (MAVLink: xxxx) for now, accept PR, and raise issue to look at errors as a separate task.

Reply:
Yeah, merge whenever. I have created https://github.com/dronecore/DroneCore/issues/127 to add eventual task to look at these messages.